sklearn/datasets/tests/test_openml.py

```diff
@@ -1572,6 +1572,31 @@ def test_fetch_openml_with_ignored_feature(monkeypatch, gzip_response, parser):
     assert "animal" not in dataset["feature_names"]
 
 
+def test_fetch_openml_strip_quotes(monkeypatch):
+    """Check that we strip the single quotes when used as a string delimiter.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/23381
+    """
+    pd = pytest.importorskip("pandas")
+    data_id = 40966
+    _monkey_patch_webbased_functions(monkeypatch, data_id=data_id, gzip_response=False)
+
+    common_params = {"as_frame": True, "cache": False, "data_id": data_id}
+    mice_pandas = fetch_openml(parser="pandas", **common_params)
+    mice_liac_arff = fetch_openml(parser="liac-arff", **common_params)
+    pd.testing.assert_series_equal(mice_pandas.target, mice_liac_arff.target)
```

Comment:
Could you please add an additional assertion on the value of `np.unique(mice_pandas.target.values)`.

Maybe could extend this test to also check the value of `.target` directly returned as a numpy array when `fetch_openml` is called with `as_frame=False`.

Reply:
I will check that the first and last values do not contain a single quote.
Indeed, we have a few lines of the original dataset and thus a single target. It will not be a nice assertion to do.